File: Presentation/Settings/SettingsWindowController.swift
import Cocoa

/// Settings window for API key configuration
class SettingsWindowController: NSWindowController {

    private var anthropicField: NSSecureTextField!
    private var groqField: NSSecureTextField!

    static func create() -> SettingsWindowController {
        let window = NSWindow(
            contentRect: NSRect(x: 0, y: 0, width: 450, height: 220),
            styleMask: [.titled, .closable],
            backing: .buffered,
            defer: false
        )
        window.title = "Settings"
        window.center()

        let controller = SettingsWindowController(window: window)
        controller.setupUI()
        return controller
    }

    private func setupUI() {
        guard let contentView = window?.contentView else { return }
        contentView.wantsLayer = true

        let padding: CGFloat = 20
        var y: CGFloat = 160

        // Anthropic API Key
        let anthropicLabel = NSTextField(labelWithString: "Anthropic API Key:")
        anthropicLabel.frame = NSRect(x: padding, y: y, width: 130, height: 20)
        contentView.addSubview(anthropicLabel)

        anthropicField = NSSecureTextField(frame: NSRect(x: 155, y: y, width: 270, height: 24))
        anthropicField.placeholderString = "sk-ant-..."
        if let key = ApiKeyManager.shared.getKey(.anthropic) {
            anthropicField.stringValue = key
        }
        contentView.addSubview(anthropicField)

        y -= 50

        // Groq API Key
        let groqLabel = NSTextField(labelWithString: "Groq API Key:")
        groqLabel.frame = NSRect(x: padding, y: y, width: 130, height: 20)
        contentView.addSubview(groqLabel)

        groqField = NSSecureTextField(frame: NSRect(x: 155, y: y, width: 270, height: 24))
        groqField.placeholderString = "gsk_..."
        if let key = ApiKeyManager.shared.getKey(.groq) {
            groqField.stringValue = key
        }
        contentView.addSubview(groqField)

        y -= 50

        // Help text
        let helpText = NSTextField(wrappingLabelWithString: "API keys are stored securely in your macOS Keychain.")
        helpText.frame = NSRect(x: padding, y: y, width: 410, height: 30)
        helpText.textColor = .secondaryLabelColor
        helpText.font = NSFont.systemFont(ofSize: 11)
        contentView.addSubview(helpText)

        // Save button
        let saveButton = NSButton(title: "Save", target: self, action: #selector(saveKeys))
        saveButton.bezelStyle = .rounded
        saveButton.frame = NSRect(x: 350, y: 15, width: 80, height: 30)
        saveButton.keyEquivalent = "\r"
        contentView.addSubview(saveButton)

        // Cancel button
        let cancelButton = NSButton(title: "Cancel", target: self, action: #selector(cancelSettings))
        cancelButton.bezelStyle = .rounded
        cancelButton.frame = NSRect(x: 260, y: 15, width: 80, height: 30)
        cancelButton.keyEquivalent = "\u{1b}"
        contentView.addSubview(cancelButton)
    }

    @objc private func saveKeys() {
        let anthropicKey = anthropicField.stringValue.trimmingCharacters(in: .whitespacesAndNewlines)
        let groqKey = groqField.stringValue.trimmingCharacters(in: .whitespacesAndNewlines)

        if !anthropicKey.isEmpty {
            _ = ApiKeyManager.shared.setKey(anthropicKey, for: .anthropic)
        }

        if !groqKey.isEmpty {
            _ = ApiKeyManager.shared.setKey(groqKey, for: .groq)
        }

        NotificationCenter.default.post(name: .apiKeysUpdated, object: nil)
        window?.close()
    }

    @objc private func cancelSettings() {
        window?.close()
    }
}

File: Presentation/Settings/SettingsWindowController.cs
using System;
using System.Linq;
using System.Windows;
using System.Windows.Controls;
using InterviewMasterApp;
using InterviewMasterApp.Domain.Model;
using InterviewMasterApp.Infrastructure.Storage;

namespace InterviewMasterApp.Presentation.Settings
{
    /// <summary>
    /// Port of SettingsWindowController.swift to WPF.
    /// Builds a settings Window in code and wires Save/Cancel logic to AppSettings and ApiKeyManager.
    /// </summary>
    public class SettingsWindowController
    {
        private Window? _window;

        // Controls
        private PasswordBox? _anthropicField;
        private PasswordBox? _groqField;
        private ComboBox? _roleDropdown;
        private ComboBox? _programmingLanguageDropdown;
        private ComboBox? _speakingLanguageDropdown;
        private TextBox? _frameworksField;

        public static Window Create()
        {
            var controller = new SettingsWindowController();
            controller.BuildWindow();
            return controller._window!;
        }

        private void BuildWindow()
        {
            _window = new Window
            {
                Title = "Settings",
                Width = 500,
                Height = 420,
                WindowStartupLocation = WindowStartupLocation.CenterScreen,
                ResizeMode = ResizeMode.NoResize
            };

            var root = new Grid { Margin = new Thickness(20) };
            _window.Content = root;

            // Define rows
            for (int i = 0; i < 12; i++) root.RowDefinitions.Add(new RowDefinition { Height = GridLength.Auto });
            root.RowDefinitions.Add(new RowDefinition { Height = new GridLength(1, GridUnitType.Star) });

            int row = 0;

            var interviewHeader = CreateSectionHeader("Interview Settings");
            Grid.SetRow(interviewHeader, row++);
            root.Children.Add(interviewHeader);

            // Role
            var rolePanel = CreateLabeledControl("Position/Role:", out _roleDropdown);
            PopulateEnumComboBox(_roleDropdown, typeof(InterviewRole), v => ((InterviewRole)v).GetDisplayName());
            SelectComboBoxItemByDisplay(_roleDropdown, AppSettings.Shared.Role.GetDisplayName());
            Grid.SetRow(rolePanel, row++);
            root.Children.Add(rolePanel);

            // Programming Language
            var progPanel = CreateLabeledControl("Programming Language:", out _programmingLanguageDropdown);
            PopulateEnumComboBox(_programmingLanguageDropdown, typeof(ProgrammingLanguage), v => ((ProgrammingLanguage)v).GetDisplayName());
            SelectComboBoxItemByDisplay(_programmingLanguageDropdown, AppSettings.Shared.ProgrammingLanguage.GetDisplayName());
            Grid.SetRow(progPanel, row++);
            root.Children.Add(progPanel);

            // Speaking Language
            var speakPanel = CreateLabeledControl("Response Language:", out _speakingLanguageDropdown);
            PopulateEnumComboBox(_speakingLanguageDropdown, typeof(SpeakingLanguage), v => ((SpeakingLanguage)v).GetDisplayName());
            SelectComboBoxItemByDisplay(_speakingLanguageDropdown, AppSettings.Shared.SpeakingLanguage.GetDisplayName());
            Grid.SetRow(speakPanel, row++);
            root.Children.Add(speakPanel);

            // Frameworks
            var frameworksPanel = CreateLabeledControl("Tech Stack:", out _frameworksField);
            _frameworksField.Text = AppSettings.Shared.Frameworks ?? string.Empty;
            _frameworksField.ToolTip = "Comma-separated list of frameworks and tools you use";
            Grid.SetRow(frameworksPanel, row++);
            root.Children.Add(frameworksPanel);

            // Divider (spacer)
            var divider = new Separator { Margin = new Thickness(0, 10, 0, 10) };
            Grid.SetRow(divider, row++);
            root.Children.Add(divider);

            // API Header
            var apiHeader = CreateSectionHeader("API Keys");
            Grid.SetRow(apiHeader, row++);
            root.Children.Add(apiHeader);

            // Anthropic Key
            var anthropicPanel = CreateLabeledPasswordBox("Anthropic API Key:", out _anthropicField);
            var anthKey = ApiKeyManager.Shared.GetKey(ApiKeyType.Anthropic);
            if (!string.IsNullOrEmpty(anthKey)) _anthropicField.Password = anthKey;
            Grid.SetRow(anthropicPanel, row++);
            root.Children.Add(anthropicPanel);

            // Groq Key
            var groqPanel = CreateLabeledPasswordBox("Groq API Key:", out _groqField);
            var groqKey = ApiKeyManager.Shared.GetKey(ApiKeyType.Groq);
            if (!string.IsNullOrEmpty(groqKey)) _groqField.Password = groqKey;
            Grid.SetRow(groqPanel, row++);
            root.Children.Add(groqPanel);

            // Help text
            var helpText = new TextBlock { Text = "API keys are stored securely in your user profile file.", FontSize = 11, Margin = new Thickness(0, 6, 0, 12) };
            Grid.SetRow(helpText, row++);
            root.Children.Add(helpText);

            // Buttons
            var buttonsPanel = new StackPanel { Orientation = Orientation.Horizontal, HorizontalAlignment = HorizontalAlignment.Right };
            var saveButton = new Button { Content = "Save", Width = 80, Margin = new Thickness(8, 0, 0, 0), IsDefault = true };
            saveButton.Click += (_, _) => SaveSettings();
            var cancelButton = new Button { Content = "Cancel", Width = 80, Margin = new Thickness(8, 0, 0, 0), IsCancel = true };
            cancelButton.Click += (_, _) => CancelSettings();
            buttonsPanel.Children.Add(cancelButton);
            buttonsPanel.Children.Add(saveButton);
            Grid.SetRow(buttonsPanel, row++);
            root.Children.Add(buttonsPanel);
        }

        private UIElement CreateSectionHeader(string title)
        {
            return new TextBlock { Text = title, FontWeight = FontWeights.Bold, FontSize = 13, Margin = new Thickness(0, 4, 0, 6) };
        }

        private FrameworkElement CreateLabeledControl(string labelText, out ComboBox comboBox)
        {
            var grid = new Grid();
            grid.ColumnDefinitions.Add(new ColumnDefinition { Width = new GridLength(160) });
            grid.ColumnDefinitions.Add(new ColumnDefinition { Width = new GridLength(1, GridUnitType.Star) });

            var label = new TextBlock { Text = labelText, VerticalAlignment = VerticalAlignment.Center };
            Grid.SetColumn(label, 0);
            grid.Children.Add(label);

            comboBox = new ComboBox { Margin = new Thickness(6, 2, 0, 2) };
            Grid.SetColumn(comboBox, 1);
            grid.Children.Add(comboBox);

            return grid;
        }

        private FrameworkElement CreateLabeledControl(string labelText, out TextBox textBox)
        {
            var grid = new Grid();
            grid.ColumnDefinitions.Add(new ColumnDefinition { Width = new GridLength(160) });
            grid.ColumnDefinitions.Add(new ColumnDefinition { Width = new GridLength(1, GridUnitType.Star) });

            var label = new TextBlock { Text = labelText, VerticalAlignment = VerticalAlignment.Center };
            Grid.SetColumn(label, 0);
            grid.Children.Add(label);

            textBox = new TextBox { Margin = new Thickness(6, 2, 0, 2) };
            Grid.SetColumn(textBox, 1);
            grid.Children.Add(textBox);

            return grid;
        }

        private FrameworkElement CreateLabeledPasswordBox(string labelText, out PasswordBox passwordBox)
        {
            var grid = new Grid();
            grid.ColumnDefinitions.Add(new ColumnDefinition { Width = new GridLength(160) });
            grid.ColumnDefinitions.Add(new ColumnDefinition { Width = new GridLength(1, GridUnitType.Star) });

            var label = new TextBlock { Text = labelText, VerticalAlignment = VerticalAlignment.Center };
            Grid.SetColumn(label, 0);
            grid.Children.Add(label);

            passwordBox = new PasswordBox { Margin = new Thickness(6, 2, 0, 2) };
            Grid.SetColumn(passwordBox, 1);
            grid.Children.Add(passwordBox);

            return grid;
        }

        private void PopulateEnumComboBox(ComboBox cb, Type enumType, Func<object, string> display)
        {
            cb.Items.Clear();
            foreach (var val in Enum.GetValues(enumType))
            {
                cb.Items.Add(new ComboBoxItem { Content = display(val), Tag = val });
            }
        }

        private void SelectComboBoxItemByDisplay(ComboBox cb, string display)
        {
            foreach (ComboBoxItem item in cb.Items)
            {
                if ((item.Content as string) == display)
                {
                    cb.SelectedItem = item;
                    return;
                }
            }
            if (cb.Items.Count > 0) cb.SelectedIndex = 0;
        }

        private void SaveSettings()
        {
            // Interview Settings
            if (_roleDropdown?.SelectedItem is ComboBoxItem roleItem && roleItem.Tag is InterviewRole role)
            {
                AppSettings.Shared.Role = role;
            }

            if (_programmingLanguageDropdown?.SelectedItem is ComboBoxItem langItem && langItem.Tag is ProgrammingLanguage pl)
            {
                AppSettings.Shared.ProgrammingLanguage = pl;
            }

            if (_speakingLanguageDropdown?.SelectedItem is ComboBoxItem spItem && spItem.Tag is SpeakingLanguage sl)
            {
                AppSettings.Shared.SpeakingLanguage = sl;
            }

            AppSettings.Shared.Frameworks = _frameworksField?.Text?.Trim() ?? string.Empty;

            // API Keys
            var anthropicKey = _anthropicField?.Password?.Trim() ?? string.Empty;
            var groqKey = _groqField?.Password?.Trim() ?? string.Empty;

            if (!string.IsNullOrEmpty(anthropicKey)) ApiKeyManager.Shared.SetKey(anthropicKey, ApiKeyType.Anthropic);
            if (!string.IsNullOrEmpty(groqKey)) ApiKeyManager.Shared.SetKey(groqKey, ApiKeyType.Groq);

            // Fire notifications (simple static events)
            AppNotifications.RaiseApiKeysUpdated();
            AppNotifications.RaiseInterviewSettingsUpdated();

            _window?.Close();
        }

        private void CancelSettings()
        {
            _window?.Close();
        }
    }

}

